### PR TITLE
GH-373: Fix random Concurrent Modification exception

### PR DESCRIPTION
**Pull request by mor-n4**
Note: since the exception is nearly impossible to reproduce, we should only merge this if the changes also make sense on their own, without considering the ConcurrentModificationException.

### CODE REVIEW

**Comment by qtran-n4** [comment]
Reconsider activating the logging disabled by @kduske-n4 when he fixed ConcurrentModificationException.

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/ExternalLibrariesReloadHelper.java [hunk 1]
```diff
@@ -69,17 +69,24 @@
 	public void reloadLibraries(final boolean refreshNpmDefinitions, final IProgressMonitor monitor)
```
Comment (qtran-n4) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/ExternalLibrariesReloadHelper.java]: According to this logics, we have to do this for ALL places in which we trigger the build.
The general question is: can multiple threads execute the build code concurrently? Probably yes at the moment.

Isn't it better to make the build code its self thread-safe with e.g. `synchronized`?

Comment (qtran-n4) [reply plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/ExternalLibrariesReloadHelper.java]: Consider using a workspace root scheduling in `ExternalLibraryBuilderHelper`.

Comment (mor-n4) [reply plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/external/ExternalLibrariesReloadHelper.java]: re 1st comment: as discussed (would not be in line with how Eclipse invokes the builder usually)

re 2nd comment: good idea! Will check that.

**Comment by mor-n4** [comment]
@yoosiba @qtran-n4
as discussed: adding a similar lock in `ExternalLibraryBuilderHelper` is a good idea, but it requires some refactoring of that and related classes which is better done in a separate task. Created #389 for that.

Builds green, merging ...

**Pull request merged**